woocommerce/woocommerce-ios
# Store creation M2: analytics

Closes: #8248 


### Description


This PR implemented the new events and properties as in pe5sF9-I3-p2#comment-1291, minus the two IAP events.

### Testing instructions


- Log out and skip onboarding if needed
- Tap `Get started`
- Tap `Log in` to log in with an account without any stores, or create an account —> it should reach the empty stores screen
- Tap the ellipsis menu at the top, and tap `Log out` —> `🔵 Tracked site_picker_logout_tapped, properties: [:]` should be shown in the console
- Tap `Add a store` to create a store —> `🔵 Tracked site_creation_step, properties: [AnyHashable("step"): "store_name”]` should be shown in the console
- Tap `Cancel` to exit the store creation flow —> `🔵 Tracked site_creation_dismissed, properties: [AnyHashable("flow"): "native", …]` should be shown in the console
- Tap `Add a store` again to create a store
- Enter some store name and tap `Continue` —> `🔵 Tracked site_creation_step, properties: [AnyHashable("step"): "domain_picker"]` should be shown in the console
- Select a domain and tap `Continue` —> `🔵 Tracked site_creation_step, properties: [AnyHashable("step"): "store_summary”]` should be shown in the console
- Tap `Continue` —> `🔵 Tracked site_creation_step, properties: [AnyHashable("step"): "plan_purchase”]` should be shown in the console
- Tap `Create Store for $/month` —> `🔵 Tracked site_creation_step, properties: [AnyHashable("step"): "web_checkout"]` should be shown in the console
- Complete the purchase —> when the purchase is successful,  `🔵 Tracked site_creation_step, properties: [AnyHashable("step"): "store_installation"]` should be shown in the console when the in-progress screen is shown
- Wait until it continues to the success screen when the site becomes a Jetpack site —> `🔵 Tracked login_woocommerce_site_created, properties: [AnyHashable("flow"): "native", AnyHashable("url"): “{{site_url”}}, AnyHashable("source"): "prologue”]` should be shown in the console
- Tap `Store Preview` —> `🔵 Tracked site_creation_site_previewed, properties: [:]` should be shown in the console
- Dismiss the webview, then tap `Manage My Store`  —> `🔵 Tracked site_creation_store_management_opened, properties: [:]` should be shown in the console

---
- [x] @jaclync tests the affected events in M1: `login_woocommerce_site_created`, `site_creation_failed`, `site_creation_dismissed`

---
